compiler-rt/lib/mpk_untrusted/alloc_site_handler.h

```diff
@@ -193,6 +206,27 @@ class AllocSiteHandler {
     }
   }
 
+  void storePidKey(pid_t threadID, PKeyInfo pkey) {
+    // Obtain map key
+    const std::lock_guard<std::mutex> lock(px);
+
+    pid_key_map.insert(std::pair<pid_t, PKeyInfo>(threadID, pkey));
+  }
+
+  llvm::Optional<PKeyInfo> getPidKey(pid_t threadID) {
+    // Obtain map key
+    const std::lock_guard<std::mutex> lock(px);
+
+    auto iter = pid_key_map.find(threadID);
+    // If PID does not contain key in map, return None.
+    if (iter == pid_key_map.end())
+      return llvm::None;
+
+    auto ret_val = iter->second;
+    pid_key_map.erase(threadID);
```

Comment:
why remove from the map? getX() APIs don't usually destroy the data they retrieve.

Reply:
Changed naming to better indicate data removal from map.

Reply:
I'm not seeing how the new name better indicates that it should be removed from the map. Yes it uses 'pop', but it also is a bit odd to have `storePKeyInfo()` and `popPendingPkeyInfo()`.  If this is intended to be a stack, then maybe it should be a stack? if the `PKeyInfo` is only used to track a pending thing, then maybe it should be `PendingPkeyInfo`, or if you want to use `PKeyInfo` elsewhere, then maybe the code here should use something like this:

 ```C++
struct PendingPKeyInfo{ 
  PKeyInfo info;
}
```

Another point is that unlike address tracking, where you need to maintain ordering between elements, there is no need for thread association. A table lookup should be fine, and might perform significantly better. If you optimize this later, you may want to consider more optimized data structures for this type of lookup.